Loop/Managers/LoopDataManager.swift

```diff
@@ -229,9 +231,47 @@ final class LoopDataManager {
             }
 
             // Invalidate cached effects affected by the override
-            self.carbEffect = nil
-            self.carbsOnBoard = nil
-            self.insulinEffect = nil
+            invalidateCachedEffects = true
+        }
+
+        if newValue.insulinSensitivitySchedule != oldValue.insulinSensitivitySchedule {
+            carbStore.insulinSensitivitySchedule = newValue.insulinSensitivitySchedule
+            doseStore.insulinSensitivitySchedule = newValue.insulinSensitivitySchedule
+            invalidateCachedEffects = true
+            analyticsServicesManager.didChangeInsulinSensitivitySchedule()
+        }
+
+        if newValue.basalRateSchedule != oldValue.basalRateSchedule {
+            doseStore.basalProfile = newValue.basalRateSchedule
+
+            if let newValue = newValue.basalRateSchedule, let oldValue = oldValue.basalRateSchedule, newValue.items != oldValue.items {
+                analyticsServicesManager.didChangeBasalRateSchedule()
+            }
+        }
+
+        if newValue.carbRatioSchedule != oldValue.carbRatioSchedule {
+            carbStore.carbRatioSchedule = newValue.carbRatioSchedule
+            invalidateCachedEffects = true
+            analyticsServicesManager.didChangeCarbRatioSchedule()
+        }
+
+        if newValue.glucoseTargetRangeSchedule?.timeZone != oldValue.glucoseTargetRangeSchedule?.timeZone {
+            analyticsServicesManager.pumpTimeZoneDidChange()
+        }
```

Comment:
All schedules should be in sync. This is some redundancy in the data model that maybe should get cleaned up at some point. But I can change which one it's checking.

Reply:
Actually, looks like this is duplicating a bit, and I'd like to change the event name from "pump" timezone to "therapy schedule" timezone change, since that covers the fact that all of them are changing.